### PR TITLE
Migrate Channel.UserBelongsToChannels to Sync by default

### PR DESCRIPTION
**Pull request by gupsho**
#### Summary
Migrate "Channel.UserBelongsToChannels" to Sync by default

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/11249

### CODE REVIEW

File: app/user.go [hunk 1]
```diff
@@ -2134,11 +2134,11 @@ func (a *App) userBelongsToTeams(userId string, teamIds []string) (bool, *model.
 }
 
 func (a *App) userBelongsToChannels(userId string, channelIds []string) (bool, *model.AppError) {
-	result := <-a.Srv.Store.Channel().UserBelongsToChannels(userId, channelIds)
-	if result.Err != nil {
-		return false, result.Err
+	result, err := a.Srv.Store.Channel().UserBelongsToChannels(userId, channelIds)
```
Comment (jespino) [inline app/user.go]: Here you can just `return a.Srv.Store.Channel().UserBelongsToChannels(userId, channelIds)`

Comment (gupsho) [reply app/user.go]: Done.

**Comment by gupsho** [comment]
> A small change needed, but otherwise looks good to me. Thanks @gupsho! ✨

Fixed.

**Comment by jespino** [comment]
Merged! Thanks @gupsho! :tada:

**Pull request merged**